cmd/release/minor/projects/projects.go

```diff
@@ -0,0 +1,133 @@
+		}
+	}
+
+	// Validate results
+	prevReleaseBranchFileCount, err := getFileCount(prevReleaseBranch)
+	if err != nil {
+		return -1, fmt.Errorf("getting previous release branch project file count: %w", err)
+	}
+	nextReleaseBranchFileCount, err := getFileCount(nextReleaseBranch)
+	if err != nil {
+		return -1, fmt.Errorf("getting next release branch project file count: %w", err)
+	}
+	if prevReleaseBranchFileCount != nextReleaseBranchFileCount {
+		return -1, fmt.Errorf("expected previous release branch file count (%d) to match next release branch file count (%d)",
+			prevReleaseBranchFileCount, nextReleaseBranchFileCount)
+	}
+	return nextReleaseBranchFileCount, nil
+}
+
+func copyDir(prevReleaseBranchPath, nextReleaseBranchPath string) error {
+	prevReleaseBranchDirs, err := os.ReadDir(prevReleaseBranchPath)
+	if err != nil {
+		return fmt.Errorf("reading directory at path %s: %w", prevReleaseBranchPath, err)
+	} else if len(prevReleaseBranchDirs) == 0 {
+		return nil
+	}
+
+	for _, prevReleaseBranchChildDir := range prevReleaseBranchDirs {
+		prevReleaseBranchChildPath := filepath.Join(prevReleaseBranchPath, prevReleaseBranchChildDir.Name())
+		nextReleaseBranchDirPath := filepath.Join(nextReleaseBranchPath, prevReleaseBranchChildDir.Name())
+		if prevReleaseBranchChildDir.IsDir() {
+			if err = os.Mkdir(nextReleaseBranchDirPath, 0755); err != nil {
+				return fmt.Errorf("creating new directory to copy %s", nextReleaseBranchDirPath)
+			}
+			if err = copyDir(prevReleaseBranchChildPath, nextReleaseBranchDirPath); err != nil {
+				return fmt.Errorf("copying child directory %s", prevReleaseBranchChildPath)
+			}
+		} else {
+			if err = copyFile(prevReleaseBranchChildPath, nextReleaseBranchDirPath); err != nil {
+				return fmt.Errorf("coping file: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+func copyFile(prevReleaseBranchFilePath, nextReleaseBranchFilePath string) error {
+	existingFile, err := os.Open(prevReleaseBranchFilePath)
+	if err != nil {
+		return err
+	}
+	defer existingFile.Close()
+
+	newFile, err := os.Create(nextReleaseBranchFilePath)
+	if err != nil {
+		return err
+	}
+	defer newFile.Close()
+
+	if filepath.Base(existingFile.Name()) == "CHECKSUMS" {
+		scanner := bufio.NewScanner(existingFile)
+		for scanner.Scan() {
+			updatedLine := strings.ReplaceAll(scanner.Text(), prevReleaseBranch, nextReleaseBranch)
+			if _, err = fmt.Fprintln(newFile, updatedLine); err != nil {
+				return fmt.Errorf("modifying CHECKSUM file: %w", err)
+			}
+		}
+		if err = scanner.Err(); err != nil {
+			return fmt.Errorf("scanning CHECKSUMS: %w", err)
+		}
+	} else {
+		if _, err = io.Copy(newFile, existingFile); err != nil {
+			return fmt.Errorf("copying files: %w", err)
+		}
+	}
+	return nil
+}
+
+func getFileCount(releaseBranch string) (int, error) {
+	pathPattern := filepath.Join(values.GetProjectPathRoot(), "*", "*", releaseBranch, "*")
+	out, err := exec.Command("bash", "-c", fmt.Sprintf("find %s -type f | wc -l", pathPattern)).Output()
```

Comment:
curious why we're shelling out here instead of using Go? not an issue but just wondering

Reply:
It seemed easier. I'm not at all sold on this approach if you'd prefer it in Go